### PR TITLE
feat: add sunrise and sunset times to seasons with interpolation

### PR DESCRIPTION
**Pull request by rayners**
Implements sunrise and sunset tracking for calendar seasons with dynamic
interpolation based on the current date's position within a season. The
feature includes UI integration across all widgets, comprehensive
default/fallback logic, and complete documentation.

**Core Features:**
- Added sunrise/sunset properties to CalendarSeason interface
- Created SunriseSunsetCalculator utility class with season interpolation
- Added sunrise/sunset values to Gregorian calendar (Baltimore, MD reference)
- Implemented getSunriseSunset() API method with full calculation logic

**Calculation & Defaults:**
- Interpolates sunrise/sunset between current and next season
- Uses Gregorian values as defaults for matching season names
- Falls back to 50/50 day/night split (sunrise at 25%, sunset at 75% of day)
- Handles year-crossing seasons (e.g., Winter spanning Dec-Feb)

**UI Integration:**
- Main widget: Displays sunrise/sunset in date display area
- Grid widget: Shows sunrise/sunset in day tooltips
- Mini widget: Toggleable sunrise/sunset row (context menu option)
- Added localization for mini widget toggle option

**Settings:**
- Added miniWidgetShowSunriseSunset client setting (default: true)
- Integrated with settings change detection for live updates

**Testing:**
- Comprehensive test suite for SunriseSunsetCalculator
- Tests cover interpolation, year-crossing, defaults, and edge cases

**Documentation:**
- Updated calendar-v1.0.0.json schema with sunrise/sunset properties
- Added regex pattern validation for HH:MM time format
- Created comprehensive Seasons section in CALENDAR-FORMAT.md
- Documented interpolation behavior and defaults
- Updated validation rules to mention optional sunrise/sunset fields

References: Baltimore, MD (39.3°N, 76.6°W) for Gregorian season values

### CODE REVIEW

File: packages/core/src/core/sunrise-sunset-calculator.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      return false;
+    }
+    if (month === startMonth && day < startDay) {
+      return false;
+    }
+    if (month === endMonth && day > endDay) {
+      return false;
+    }
+    return true;
+  }
```
Comment (rayners) [inline packages/core/src/core/sunrise-sunset-calculator.ts]: I think this is implemented already in a utility module of some sort. Look for a pre-existing version of this function and use that instead.

File: packages/core/src/core/sunrise-sunset-calculator.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+  /**
+   * Get total days in a year for the calendar
+   */
+  private static getDaysInYear(calendar: SeasonsStarsCalendar): number {
+    if (!calendar.months) {
+      return 365;
+    }
+    return calendar.months.reduce((sum, month) => sum + (month.days ?? 30), 0);
+  }
```
Comment (rayners) [inline packages/core/src/core/sunrise-sunset-calculator.ts]: I'm certain this is implemented already

File: packages/core/src/core/sunrise-sunset-calculator.ts [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+  /**
+   * Convert hours as decimal to time string (HH:MM)
+   */
+  static hoursToTimeString(hours: number): string {
+    const h = Math.floor(hours);
+    const m = Math.round((hours - h) * 60);
+    return `${h.toString().padStart(2, '0')}:${m.toString().padStart(2, '0')}`;
+  }
+}
```
Comment (rayners) [inline packages/core/src/core/sunrise-sunset-calculator.ts]: Use pre-existing implementations if they exist.

File: packages/core/src/types/calendar.d.ts [hunk 4]
```diff
@@ -195,6 +195,18 @@ export interface CalendarSeason {
...
   icon?: string;
   color?: string;
+  /**
+   * Sunrise time on the first day of the season in 24-hour format (HH:MM).
+   * If not specified, defaults will be calculated based on season name matching
+   * with Gregorian calendar or interpolated from surrounding seasons.
+   */
+  sunrise?: string;
+  /**
+   * Sunset time on the first day of the season in 24-hour format (HH:MM).
```
Comment (rayners) [inline packages/core/src/types/calendar.d.ts]: It's not 24-hour format, it's simply "HH:MM". Not all calendars have 24 hour days.

**Comment by rayners** [comment]
@claude We don't need fallback code for when the engine isn't available. That's just extra code to test that isn't necessary. Always assume the engine instance is available and fail fast if it isn't.
